eureka-core/src/main/java/com/netflix/eureka/cluster/HttpReplicationClient.java

```diff
@@ -15,4 +16,5 @@
 
     EurekaHttpResponse<ReplicationListResponse> submitBatchUpdates(ReplicationList replicationList);
 
+    default void addReplicationClientFilter(ClientFilter clientFilter) {};
```

Comment:
Adding this method here would force a Jersey dependency on HttpReplicationClient.  You should be able to provide your own PeerEurekaNodes implementation similar to Jersey2PeerEurekaNodes that would add filters to the JerseyReplicationClient.

Reply:
So no need for this change, but the one below is still valid, correct?

Reply:
Given the static nature of these API's I'd say yes.  Ideally I'd rather have a proper factory and only allow adding filters at creation time.  Not sure about the thread safety guarantees when adding a filter to an existing client.  Not sure it's worth the effort to refactor that now.

Reply:
A new `PeerEurekaNodes` implementation itself isn't enough to add filters to `JerseyReplicationClient` because `JerseyReplicationClient.createReplicationClient()` returns `HttpReplicationClient`, that's why I went ahead and changed the interface. But I agree that it's not great to force a Jersey dependency on that client. 

So maybe I can try implementing the second option proposed in the issue, which is passing in additional filters on the static creation method, or use `ReplicationClientOptionalArgs` if we follow the pattern of DiscoveryClient using `DiscoveryClientOptionalArgs`.

Reply:
JerseyReplicationClient.createReplicationClient() returns a JerseyReplicationClient and not HttpReplicationClient.
 https://github.com/Netflix/eureka/blob/07876a70a547ea52822c286b49c001971a1e6880/eureka-core/src/main/java/com/netflix/eureka/transport/JerseyReplicationClient.java#L135

Reply:
We don't interact directly with that. We have a bean that extends `PeerEurekaNodes`. I guess we'd override `createPeerEurekaNode()`?

Reply:
RE: concurrency, we would only add a client filter before eureka server starts up during the spring lifecycle.

Reply:
@elandau I updated this PR removing the `addReplicationClientFilter` method from the interface.